src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,37 +1,104 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+interface Movie {
+  title: string;
+  description: string;
+  imgUrl: string;
+  imdbUrl: string;
+  imdbId: string;
+}
+
+interface NewMovieProps {
+  onAddMovie: (newMovie: Movie) => void;
+}
+
+export const NewMovie = ({ onAddMovie }: NewMovieProps) => {
+  const [count, setCount] = useState(0);
+
+  const [formData, setFormData] = useState<Movie>({
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  });
+
+  const handleChange = (fieldName: keyof Movie, newValue: string) => {
+    setFormData(prevState => ({ ...prevState, [fieldName]: newValue }));
+  };
```

Comment:
you can use event.target properties
```suggestion
  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
    const { name, value } = e.target;

    setNewMovie(prevMovie => ({ ...prevMovie, [name]: value }));
  };
```